jbig2/jbig2_compress_image.go

```diff
@@ -0,0 +1,99 @@
+/*
+ * This example showcases the conversion of the jpg encoded image
+ * into jbig2 encoding format.
+ *
+ * The result jbig2 file is compressed with a lossless method stored with
+ * the standard  jbig2 - .jb2 extension. The file compressed in this example
+ * is stored in just 377 bytes - compared to the 142 794 bytes for original image.
+ * This gives compression ratio (uncompressed size/compressed size) of 378.76, which leads to 99.735%
+ * space savings for given example.
+ */
+
+package main
+
+import (
+	"fmt"
+	"image"
+	// load jpeg decoder
+	_ "image/jpeg"
+	"os"
+
+	"github.com/unidoc/unipdf/v3/core"
+)
+
+func main() {
+	// let's read an jpeg rgba image from the file
+	// convert it into JBIG2Image using auto threshold
+	// and compress the black and white result into another file
+	// read an image file 'my-image.jpg'
+	f, err := os.Open("checkerboard-squares-black-white.jpg")
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+		os.Exit(1)
+	}
+	defer f.Close()
+
+	// decode the jpeg image.
+	img, _, err := image.Decode(f)
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	// JBIG2Encoder requires core.JBIG2Image as an input.
+	// In order to convert golang image.Image into core.JBIG2Image use
+	// a core.GoImageToJBIG2 function.
```

Comment:
`the core.GoImageToJBIG2 function` instead of `a  core.GoImageToJBIG2`.

Reply:
Done.